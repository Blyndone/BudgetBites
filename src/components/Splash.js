import * as React from 'react';
import { View, StyleSheet, SafeAreaView, Image } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { Button, Text } from 'react-native-paper';

const Splash = ({ navigation }) => {
  return (
    <SafeAreaView style={styles.container}>
      <Image
        source={require('../../assets/BB-logo.png')}
        style={{
          width: 300,
          height: 300,
        }}
      />
      <Text style={styles.titleText}>Welcome to Budget Bites!</Text>
      <View>
        <View style={styles.buttoncontainer}>
          <Button
            mode="contained"
            title="Login"
            buttonColor="#eb6b34"
            labelStyle={{ fontSize: 16, color: 'black' }}
            onPress={() => navigation.navigate('Login')}
          >
            Login
          </Button>
        </View>
        <View style={styles.buttoncontainer}>
          <Button
            mode="contained"
            title="CreateAccount"
            buttonColor="#eb6b34"
            labelStyle={{ fontSize: 16, color: 'black' }}
            onPress={() => navigation.navigate('Account Creation')}
          >
            Create Account
          </Button>
        </View>
        <View style={styles.buttoncontainer}>
          <Button
            mode="contained"
            title="Guest Login"
            buttonColor="#eb6b34"
            labelStyle={{ fontSize: 16, color: 'black' }}
            onPress={() => navigation.navigate('Guest Main View')}
          >
            Guest Login
          </Button>
        </View>
        {/* <View style={styles.buttoncontainer}>
          <Button
            mode="contained"
            title="Guest Login"
            buttonColor="#eb6b34"
            labelStyle={{ fontSize: 16, color: 'black' }}
            onPress={() => {
              data = { user_id: 1, user_name: 'customer', user_zip: 12345 };
              navigation.navigate({
                name: 'Buyer Main View',
                params: { data },
              });
            }}
          >
            customer
          </Button>
        </View>
        <View style={styles.buttoncontainer}>
          <Button
            mode="contained"
            title="Guest Login"
            buttonColor="#eb6b34"
            labelStyle={{ fontSize: 16, color: 'black' }}
            onPress={() => {
              data = { user_id: 1, user_name: 'customer' };
              navigation.navigate({
                name: 'Seller Main View',
                params: { data },
              });
            }}
          >
            Seller
          </Button>
        </View> */}
        {/* <Button
          mode="contained"
          title="Create Listing"
          buttonColor="#eb6b34"
          onPress={() => navigation.navigate('Seller Create Listing')}
        >
          Create Listing
        </Button> */}
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'teal',
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttoncontainer: { margin: 10 },
  titleText: {
    fontSize: 50,
    textAlign: 'center',
    color: 'black',
    fontWeight: 'bold',
    fontFamily: 'Helvetica',
  },
  button: {
    color: '#f194ff',
    backgroundColor: '#f194ff',
  },
  separator: {
    marginVertical: 8,
    borderBottomColor: '#737373',
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
});

export default Splash;
