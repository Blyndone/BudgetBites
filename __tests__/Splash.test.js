import React from 'react';
import renderer from 'react-test-renderer';
import Splash from '../src/components/Splash';

test('renders correctly', () => {
  const tree = renderer.create(<Splash />).toJSON();
  expect(tree).toMatchSnapshot();
});
