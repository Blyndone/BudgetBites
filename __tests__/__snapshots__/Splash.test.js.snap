// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "teal",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 300,
        "width": 300,
      }
    }
  />
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#00b3b3",
        "borderRadius": 20,
        "marginVertical": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 10,
        "width": "90%",
      }
    }
    testID="surface-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#00b3b3",
          "borderRadius": 20,
          "flex": undefined,
          "padding": 8,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 3,
        }
      }
      testID="surface"
    >
      <View>
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                {
                  "color": "black",
                  "fontFamily": "sans-serif",
                  "fontSize": 55,
                  "fontStyle": "italic",
                  "fontWeight": "bold",
                  "marginBottom": 30,
                  "textAlign": "center",
                },
              ]
            }
          >
            Welcome to
            

             Budget Bites!
             
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "margin": 10,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "#eb6b34",
                "borderRadius": 20,
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "width": "50%",
              }
            }
            testID="button-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#eb6b34",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "minWidth": 64,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="button-container"
              title="Login"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          [
                            {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            false,
                            {
                              "color": "rgba(255, 255, 255, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "color": "black",
                              "fontSize": 16,
                            },
                          ],
                        ],
                      ]
                    }
                    testID="button-text"
                  >
                    Login
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "#eb6b34",
                "borderRadius": 20,
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "width": "50%",
              }
            }
            testID="button-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#eb6b34",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "minWidth": 64,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="button-container"
              title="CreateAccount"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          [
                            {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            false,
                            {
                              "color": "rgba(255, 255, 255, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "color": "black",
                              "fontSize": 16,
                            },
                          ],
                        ],
                      ]
                    }
                    testID="button-text"
                  >
                    Create Account
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "#eb6b34",
                "borderRadius": 20,
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "width": "50%",
              }
            }
            testID="button-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#eb6b34",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "minWidth": 64,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="button-container"
              title="Guest Login"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          [
                            {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            false,
                            {
                              "color": "rgba(255, 255, 255, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "color": "black",
                              "fontSize": 16,
                            },
                          ],
                        ],
                      ]
                    }
                    testID="button-text"
                  >
                    Guest Login
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
